frontend/javascripts/oxalis/model_initialization.js

```diff
@@ -650,5 +655,36 @@ function applyLayerState(stateByLayer: UrlStateByLayer) {
         }
       }
     }
+
+    if (layerState.meshInfo) {
+      const { meshFileName: currentMeshFileName, meshes } = layerState.meshInfo;
+
+      if (currentMeshFileName != null) {
+        Store.dispatch(updateCurrentMeshFileAction(effectiveLayerName, currentMeshFileName));
+      }
+
+      for (const mesh of meshes) {
+        const { segmentId, seedPosition } = mesh;
+        if (mesh.isPrecomputed) {
+          const { meshFileName } = mesh;
+          Store.dispatch(
+            loadPrecomputedMeshAction(segmentId, seedPosition, meshFileName, effectiveLayerName),
+          );
+        } else {
+          const { mappingName, mappingType } = mesh;
+          Store.dispatch(
+            loadAdHocMeshAction(
+              segmentId,
+              seedPosition,
+              // `undefined` values will be interpreted to mean the currently active mapping which
+              // is not desired for meshes loaded via url.
```

Comment:
This seems a bit dangerous. Maybe the action creator should accept an optional `mappingInfo` object? If `mappingInfo == null`, the currently active mapping (if it exists) is used. If `mappingInfo` is defined, its content is used (which also allows `{mappingName: undefined, mappingName: undefined}` to enforce that no mapping is used). What do you think?

Reply:
That's a very good idea and the code looks much better now :)